dcpy/test/library/test_config.py

```diff
@@ -53,3 +57,32 @@ def test_backwards_compatility_with_jinja_version():
     config = Config(get_config_file("bpl_libraries_sql_deprecated"))
     computed = config.compute
     assert computed.version == config.version_today
+
+
+def test_url_with_override_path():
+    config = Config(get_config_file("url"), path_override=FAKE_PATH)
+    dataset = config.compute
+    assert dataset.source.url
+    assert dataset.source.url.path == FAKE_PATH
+    assert dataset.source.gdalpath == f"{FAKE_PATH}/{dataset.source.url.subpath}"
+
+
+@patch("dcpy.library.script.bpl_libraries.Scriptor.runner")
```

Comment:
I'm quite intrigued by using `patch`. Any rec how to better understand its scope?

Reply:
patch is my new favorite `mock` function.

It uses MagicMock, but the MagicMock that existed in this file before actually affected other tests in unrelated files (edm-recipes tests were failing because their calls were mocked)

This annotation (and then argument expected for the test function is equivalent to
```
def test_script_with_override_path():
    with patch("dcpy.library.script.bpl_libraries.Scriptor.runner") as runner:
        runner.return_value = FAKE_PATH
        config = Config(get_config_file("bpl_libraries_sql"), path_override=FAKE_PATH)
        dataset = config.compute
        assert dataset.source.script
        assert dataset.source.script.path == FAKE_PATH
        assert dataset.source.gdalpath == FAKE_PATH
```

So the scope is just within this test. Within that `with` block, all calls to the patched function are mocked

Reply:
There's some argument to using the latter form - a little clearer what the function is rather than having this "assumed" argument for the test

Reply:
thanks for the example! Makes sense. Do you know if you can use patch with pytest parametrize feature?

Reply:
I would assume so, just keep in mind that annotations sort of go "inside-out". So if you had

```python
@patch(myfunc)
@pytest.mark.paramaterize("a, b, c", [...])
test_something(a, b, c, myfunc):
    ...
```

the order is important - paramaterization is going to be the first wrapper, so its dedicated args go first, and then after that the parametrized functions are patched, so the patch implied argument goes second. If you patched and parametrized in opposite order, it'd be `myfunc, a, b, c` for args.